### DISCUSSION TITLE
Does this speech-to-text conversion work offline

### DISCUSSION THREAD

**Discussion by problemSolvingProgramming** [coqui-ai/STT#1799] [Q&A] [answered]
Does this speech-to-text conversion work offline? Or is it online? Is there a program in C # to say something through the microphone to test this model? How does it work?

**Comment by reuben** [comment, marked as answer]
Yes, with the package installed and the model files downloaded, the speech-to-text process can happen fully offline. We have an API reference for the .NET package [here](https://stt.readthedocs.io/en/latest/DotNet-API.html) and some examples [here](https://github.com/coqui-ai/STT-examples#windowsc). Note that for now you'll have to build the .NET package yourself. We're working to get all the language and platform support up and running as quickly as possible, including .NET on Windows.

**Reply by problemSolvingProgramming** [reply]
@reuben Thanks . One problem I see is that most of the training given to speech models is related to general words such as hello and ...
If we want to use specialized terms for Weiss comment in the industry, we will have a problem. Is it possible for you to make a model for industrial automation, including oil and gas?

**Reply by reuben** [reply]
General models can be specialized through a number of methods such as [fine tuning](https://stt.readthedocs.io/en/latest/TRAINING.html#continuing-training-from-a-release-model), [using a constrained language model](https://stt.readthedocs.io/en/latest/LANGUAGE_MODEL.html), or using the [hot word boosting](https://stt.readthedocs.io/en/latest/HotWordBoosting-Examples.html) feature. If you train such a model we'd be happy to host it/feature it on READMEs/documentation, etc.

**Reply by Emekaborisama** [reply]
The constrained language URL is invalid  https://stt.readthedocs.io/en/latest/Scorer.html

**Reply by reuben** [reply]
Thanks! The docs have been updated since I replied. The new page is here: https://stt.readthedocs.io/en/latest/LANGUAGE_MODEL.html

**Reply by reuben** [reply]
There is also the [scorer part of the STT Playbook](https://stt.readthedocs.io/en/latest/playbook/SCORER.html) which is relevant here.
